dashboard-app/src/lib/ui/components/Boundaries/LoadingBoundary.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import ErrorBoundary from './ErrorBoundary';
+import { ColoursEnum } from '../../design_system';
+import { BeatLoader } from 'react-spinners';
+
+
+type LoadingBoundaryProps = {
+  isLoading?: boolean | (() => boolean)
+};
+
+
+class LoadingBoundary extends ErrorBoundary<LoadingBoundaryProps> {
```

Comment:
Why does this extend `ErrorBoundary`? Does it need to use the `componentDidCatch` hook?

Reply:
So that it acts as a boundary for errors as well as for loading states. If it doesn't extend `ErrorBoundary` I'd just have to re-write all that functionality again.

Reply:
Ah, I got confused with naming! I assumed `ThrowIfError` was the error boundary. Is there a reason that this components functionality is separate to`LoadingBoundary`?

Reply:
Because an error boundary component cannot catch errors that it throws itself, the error must be thrown by a child component.

Reply:
Is the aim to keep `ThrowIfError` indefinitely or hand over throwing responsibilities to internal function calls?

Reply:
I think we need to keep it, but it should be internal to the boundary components and not used directly by any other components.